encoding/varint_test.ts

```diff
@@ -0,0 +1,97 @@
+import { decodeU32, decodeU64, encodeU32, encodeU64 } from "./varint.ts";
+import { assertEquals } from "../testing/asserts.ts";
+Deno.test({
+  name: "VarInt encode",
+  fn: function () {
+    assertEquals(encodeU32(0), new Uint8Array([0]));
+    assertEquals(encodeU32(1), new Uint8Array([1]));
+    assertEquals(encodeU32(127), new Uint8Array([127]));
+    assertEquals(encodeU32(128), new Uint8Array([128, 1]));
+    assertEquals(encodeU32(255), new Uint8Array([255, 1]));
+    assertEquals(encodeU32(25565), new Uint8Array([221, 199, 1]));
+    assertEquals(encodeU32(2097151), new Uint8Array([255, 255, 127]));
+    assertEquals(
+      encodeU32(2147483647),
+      new Uint8Array([255, 255, 255, 255, 7]),
+    );
+    assertEquals(encodeU32(-1), new Uint8Array([255, 255, 255, 255, 15]));
```

Comment:
Are negative values considered valid input?

Reply:
No, gonna add a range check for that